arrow/arrow.py

```diff
@@ -1210,36 +1211,24 @@ def is_between(self, start, end, bounds="()"):
 
         if not isinstance(start, Arrow):
             raise TypeError(
-                f"Cannot parse start date argument type of '{type(start)}'."
+                f"Cannot parse start date argument type of {type(start)!r}."
             )
 
         if not isinstance(end, Arrow):
-            raise TypeError(f"Cannot parse end date argument type of '{type(start)}'.")
+            raise TypeError(f"Cannot parse end date argument type of {type(start)!r}.")
 
         include_start = bounds[0] == "["
         include_end = bounds[1] == "]"
 
-        target_timestamp = self.float_timestamp
-        start_timestamp = start.float_timestamp
-        end_timestamp = end.float_timestamp
+        target_ts = self.float_timestamp
+        start_ts = start.float_timestamp
+        end_ts = end.float_timestamp
 
-        if include_start and include_end:
-            return (
-                target_timestamp >= start_timestamp
-                and target_timestamp <= end_timestamp
-            )
-        elif include_start and not include_end:
-            return (
-                target_timestamp >= start_timestamp and target_timestamp < end_timestamp
-            )
-        elif not include_start and include_end:
-            return (
-                target_timestamp > start_timestamp and target_timestamp <= end_timestamp
-            )
-        else:
-            return (
-                target_timestamp > start_timestamp and target_timestamp < end_timestamp
-            )
+        return (
+            (start_ts <= start_ts <= end_ts)
```

Comment:
Won't this always evaluate to true since `start_ts <= start_ts` will always be true?

Reply:
Thank you, @jadchaar, for pointing that out. That was my silly mistake. I prepared and tested everything locally and then shortened the variable names and that was it. I have fixed it and repushed.
Maybe we should extend the tests to catch such issues.

Reply:
If you would like to go for it, but looking at the `IsBetween` test cases, they seem pretty comprehensive, so I am at a loss as to why they did not catch this :(.

Reply:
This test does exactly the job:

```python
   def test_target_before_start_inclusive(self):
       target = arrow.Arrow.fromdatetime(datetime(2020, 12, 24))
       start = arrow.Arrow.fromdatetime(datetime(2020, 12, 25))
       end = arrow.Arrow.fromdatetime(datetime(2020, 12, 26))
       result = target.is_between(start, end, "[]")
       assert not result
```

but `flake8` does not let me commit this new test because of the unrelated:

```
tests/test_arrow.py:359:13: B015 Pointless comparison. This comparison does nothing but wastes CPU instructions. Remove it.
tests/test_arrow.py:367:13: B015 Pointless comparison. This comparison does nothing but wastes CPU instructions. Remove it.
tests/test_arrow.py:380:13: B015 Pointless comparison. This comparison does nothing but wastes CPU instructions. Remove it.
tests/test_arrow.py:388:13: B015 Pointless comparison. This comparison does nothing but wastes CPU instructions. Remove it.
```

Reply:
Hi @eumiro, thanks for bring this up. How are these flake8 errors being invoked? I am unable to see this warnings when I run `tox -e lint`.

Reply:
```python
❯ tox -e lint
lint installed: appdirs==1.4.4,cfgv==3.2.0,distlib==0.3.1,filelock==3.0.12,identify==1.5.10,nodeenv==1.5.0,pre-commit==2.9.3,PyYAML==5.3.1,six==1.15.0,toml==0.10.2,virtualenv==20.2.2
lint run-test-pre: PYTHONHASHSEED='3833388651'
lint run-test: commands[0] | pre-commit install
pre-commit installed at .git/hooks/pre-commit
lint run-test: commands[1] | pre-commit run --all-files --show-diff-on-failure
Trim Trailing Whitespace.................................................Passed
Fix End of Files.........................................................Passed
Fix python encoding pragma...............................................Passed
Fix requirements.txt.....................................................Passed
Check python ast.........................................................Passed
Check Yaml...............................................................Passed
Check for case conflicts.................................................Passed
Check docstring is first.................................................Passed
Check for merge conflicts................................................Passed
Debug Statements (Python)................................................Passed
isort....................................................................Passed
pyupgrade................................................................Passed
check for eval().........................................................Passed
check blanket noqa.......................................................Passed
rst ``code`` is two backticks............................................Passed
black....................................................................Passed
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

tests/test_arrow.py:358:13: B015 Pointless comparison. This comparison does nothing but wastes CPU instructions. Remove it.
tests/test_arrow.py:366:13: B015 Pointless comparison. This comparison does nothing but wastes CPU instructions. Remove it.
tests/test_arrow.py:379:13: B015 Pointless comparison. This comparison does nothing but wastes CPU instructions. Remove it.
tests/test_arrow.py:387:13: B015 Pointless comparison. This comparison does nothing but wastes CPU instructions. Remove it.

ERROR: InvocationError for command /home/miro/src/arrow/.tox/lint/bin/pre-commit run --all-files --show-diff-on-failure (exited with code 1)
_______________________________________________________________ summary ________________________________________________________________
ERROR:   lint: commands failed
```

Reply:
Ah ok turns out I had a stale pre-commit cache. I think flake8 bugbear was updating, but the old pre-commit cache was being used since the main flake8 version did not change.

I am gonna push a PR to fix this, and I will add your test while I am at it :).